sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/catalog/Catalog.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.sql.meta.catalog;
+
+import java.util.Map;
+import org.apache.beam.sdk.extensions.sql.meta.store.MetaStore;
+
+public interface Catalog {
+  String type();
+
+  MetaStore metaStore();
+
+  String name();
+
+  Map<String, String> properties();
+
+  void initialize(String name, Map<String, String> properties);
```

Comment:
Can we avoid having a stateful API with a non-initialized state? This method / these parameters seem like they should be parameters of a static factory method and/or factory object.

UPDATE: I just noticed this is the same as the `create` method of the `CatalogManager` so we should not have this method here.